synapse/storage/engines/sqlite.py

```diff
@@ -105,6 +107,14 @@ def server_version(self):
         """
         return "%i.%i.%i" % self.module.sqlite_version_info
 
+    def in_transaction(self, conn: Connection) -> bool:
+        return conn.in_transaction  # type: ignore
+
+    def set_autocommit(self, conn: Connection, autocommit: bool):
+        # Twisted doesn't let us set attributes on the connections, so we can't
+        # set the connection to autocommit mode.
```

Comment:
what's Twisted to do with it?

Reply:
The connections we get are Twisted connections that wrap the underlying native connections, the wrapper implements `__getattr__` but not `__setattr__` so we can't set the autocommit flag

Reply:
oh, right.

I suppose you could delve into `conn._connection`, but I guess it doesn't really matter? It might be worth commenting somewhere (at the call site, maybe), that this is will only *attempt* to enable autocommit, so the transaction must do an explicit commit anyway?

Reply:
Yeah, we could go and pull out `_connection`, but then I'd want to handle the case where they rename that variable in a future release.